src/python/dxpy/dx_extract_utils/InputsValidator.py

```diff
@@ -174,100 +178,92 @@ def interpret_mutually_exclusive_group(self, check):
-#                 % self.http_request["recordTypes"]
-#             )
-
-#     def get_http_request_info(self):
-#         return self.http_request
+class PathValidator:
+    """
+    PathValidator class checks for invalid object inputs and its combination with passed arguments.
+    """
+
+    def __init__(self, parser_dict, project, entity_result, error_handler=print):
+        self.parser_dict = parser_dict
+        self.project = project
+        self.entity_result = entity_result
+        self.error_handler = error_handler
+
+        self.record_http_request_info = None
+
+    def throw_error(self, message):
+        self.error_handler(message)
+
+    def try_populate_record_http_request_info(self):
+        # record_http_request_info contains crucial information for records
+        try:
+            self.record_http_request_info = DXHTTPRequest(
+                "/" + self.entity_result["id"] + "/visualize",
+                {"project": self.project, "cohortBrowser": False},
+            )
+        except (InvalidInput, InvalidState):
+            self.throw_error(
+                "Invalid cohort or dataset: {}".format(self.entity_result["id"]),
+            )
+        except Exception as details:
+            self.throw_error(str(details))
+
+    def assure_object_found(self):
+        # for object in a different project:
+        if self.project != self.entity_result["describe"]["project"]:
+            self.throw_error(
+                'Unable to resolve "{}" to a data object or folder name in {}. Please make sure your object is in your selected project.'.format(
+                    self.parser_dict.get("path"), self.project
+                )
+            )
+
+    def assure_cohort_or_dataset(self):
+        # resolving non record/cohort type
+        if self.entity_result is None:
+            self.throw_error(
+                "The path must point to a record type of cohort or dataset, not a folder."
+            )
+
+        if self.entity_result["describe"]["class"] != "record":
+            self.throw_error(
+                "Invalid path. The path must point to a record type of cohort or dataset and not a {} object.".format(
+                    self.entity_result["describe"]["class"]
+                )
+            )
+
+        # since object is record:
+        self.try_populate_record_http_request_info()
+        if not (
+            ("Dataset" in self.record_http_request_info["recordTypes"])
+            or ("CohortBrowser" in self.record_http_request_info["recordTypes"])
+        ):
+            self.throw_error(
+                "Invalid path. The path must point to a record type of cohort or dataset and not a {} object."
+            ).format(self.record_http_request_info["recordTypes"])
+
+    def assure_dataset_version(self):
+        # checking cohort/dataset version
+        dataset_version = float(self.record_http_request_info["datasetVersion"])
+        if dataset_version < 3.0:
```

Comment:
I'm not sure but this might need to be explicitly `if dataset_version != 3.0`

Reply:
I also saw it in preview code, as you mentioned but as I was going through cliexpress spec errors and messaging I found "The record id or path is a cohort or dataset but the version is less than 3.0." My error message should in fact match it. Thanks for catching this.